### DISCUSSION TITLE
How to get result after pop

### DISCUSSION THREAD

**Discussion by ebelevics** [csells/go_router#264] [Q&A]
I started to migrate Navigator 1.0 to this package, but I stumbled with this issue. In 1.0 you get Future<T?> after pushed screen is poped with `onPressed: () async {final results = await Navigator.pushNamed(`. How do you achieve this with GoRouter? 

```dart
Future<T?> pushNamed<T extends Object?>(
  BuildContext context,
  String routeName, {
  Object? arguments,
})
```

**Comment by csells** [comment]
hopefully these docs answer that question: https://gorouter.dev/navigator-integration

**Reply by Christian-Martensson** [reply]
The link https://gorouter.dev/navigator-integration is broken, is the documentation still available somewhere? Thank you

**Reply by csells** [reply]
> The link [gorouter.dev/navigator-integration](https://gorouter.dev/navigator-integration) is broken, is the documentation still available somewhere? Thank you

Yes. These docs have been moved here: https://gorouter.dev/user-input

Where did you see the old link?

**Reply by Christian-Martensson** [reply]
Sorry about the delay. The old link is from origin of this discussion. I originally replied to this comment from you: 

"hopefully these docs answer that question: https://gorouter.dev/navigator-integration"

Thank you,
Chris

**Comment by ebelevics** [comment]
Thank you for providing example. Yes, the Navigator.push example did the trick, but it seems it's the only way how I can use results in previous page. GoRouter example doesn't seem to provide `setState(() => widget.family.people.add(person));` to update state which was present in Navigator.push example. Would be nice tho to have toPushNamedResults for mobile specific case.

**Comment by csells** [comment]
You don't need that call to `setState()` in the GR example. It works just fine w/o it.
